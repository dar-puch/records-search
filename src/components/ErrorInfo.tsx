import * as React from 'react';
import { Card, CardHeader, CardContent } from '@material-ui/core';
interface IErrorInfo {
  error: string;
}

export const ErrorInfo: React.FC<IErrorInfo> = ({ error }) => {

  if (isNaN(+error)) {
    return (
      <Card style={{ width: '512px' }}>
        <CardHeader style={{ color: '#fff', height: '176px', background: 'url(../assets/images/flower.jpg) center / cover' }}>No internet</CardHeader>
        <CardContent>
          Check your connection
    </CardContent>
      </Card>
    );
  }
  return (
    <div>
      <img src={`https://http.cat/${error}`} alt={error} />
    </div>
  );
};
