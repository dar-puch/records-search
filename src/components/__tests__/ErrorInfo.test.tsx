import * as React from "react";
import { shallow, mount } from "enzyme";
import { ErrorInfo } from "../ErrorInfo";
import { CardContent } from "@material-ui/core";

describe("<ErrorInfo/>", () => {
  it("renders correctly", () => {
    const wrapper = mount(<ErrorInfo error="350" />);
    expect(wrapper).toMatchSnapshot();
  });
  it("displays proper image from httpcat when error is numeric", () => {
    const wrapper = mount(<ErrorInfo error="350" />);
    expect(wrapper.find("img").prop("src")).toMatch(
      /(https:\/\/http\.cat\/350)/
    );
  });
  it("displays 'no internet' card when error is not numeric", () => {
    const wrapper = mount(<ErrorInfo error="problem" />);
    expect(wrapper.find(CardContent).text()).toEqual("Check your connection");
  });
});
