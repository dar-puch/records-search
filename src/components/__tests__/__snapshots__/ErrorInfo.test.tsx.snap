// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorInfo/> renders correctly 1`] = `
<Component
  error="350"
>
  <div>
    <img
      src="https://http.cat/350"
    />
  </div>
</Component>
`;
